File: 1954-replace-all-digits-with-characters/replace-all-digits-with-characters.cpp
class Solution {
public:
    string replaceDigits(string s) {
        string ans="";
        ans+=s[0];
        for(int i=1;i<s.size();i++)
            if(isdigit(s[i])){
                char ch=s[i-1];
                int n=s[i]-'0';
                ch+=n;
                ans+=ch;
            }
            else
                ans+=s[i];
        return ans;
    }
};

static const int __ = [](){
    ios_base::sync_with_stdio(false);
    cin.tie(NULL);
    cout.tie(NULL);
    return 0;
}();

File: 1954-replace-all-digits-with-characters/replace-all-digits-with-characters.java
class Solution {
    public String replaceDigits(String s) {
        StringBuilder ans = new StringBuilder();
        ans.append(s.charAt(0));
        for (int i = 1; i < s.length(); i++) {
            if (Character.isDigit(s.charAt(i))) {
                char ch = s.charAt(i - 1);
                int n = s.charAt(i) - '0';
                ch += n;
                ans.append(ch);
            } else {
                ans.append(s.charAt(i));
            }
        }
        return ans.toString();
    }
}